ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -587,6 +674,26 @@ public static short getVersion(String idStr) throws IOException {
     CRAFTED
   }
 
+  /**
+   * Indicates the type of blob this is.
+   */
+  public enum BlobDataType {
+    /**
+     * Indicates a fully contained blob.
+     */
+    SIMPLE,
+
+    /**
+     * Indicates a composite (metadata) blob.
+     */
+    COMPOSITE,
```

Comment:
In most of the comments,  `COMPOSITE` is assumed the whole blob(meta + data). 
`METADATA` may be more appropriate here.

Reply:
This was copied from NotificationBlobType (as opposed to simply creating a member variable of that enum).  Shouldn't they be consistent?  Why do we use Composite in the existing enum?